libvast/vast/fwd.hpp

```diff
@@ -64,6 +64,18 @@ struct v0;
 
 } // namespace fbs
 
+// -- enums --------------------------------------------------------------------
+
+/// The possible encodings of a table slice.
+/// @note This encoding is unversioned. Newly created table slices are
+/// guaranteed to use the newest vesion of the encoding, while deserialized
+/// table slices may use an older version.
+enum class table_slice_encoding : uint8_t {
```

Comment:
Can you move the definition to its own file and keep only the declaration in here?

There also is an enum category already afaik, no need to open a new one.